doc/api/packages.md

```diff
@@ -274,41 +275,49 @@ a project that previous exported `main`, `lib`,
 }
 ```
 
-Alternatively a project could choose to export entire folders:
+Alternatively a project could choose to export entire folders both with and
+without extensioned subpaths using export patterns:
 
 ```json
 {
-  "name": "my-mod",
+  "name": "my-package",
   "exports": {
     ".": "./lib/index.js",
     "./lib": "./lib/index.js",
     "./lib/*": "./lib/*.js",
+    "./lib/*.js": "./lib/*.js",
     "./feature": "./feature/index.js",
     "./feature/*": "./feature/*.js",
+    "./feature/*.js": "./feature/*.js",
     "./package.json": "./package.json"
   }
 }
 ```
 
-As a last resort, package encapsulation can be disabled entirely by creating an
-export for the root of the package `"./*": "./*"`. This exposes every file
-in the package at the cost of disabling the encapsulation and potential tooling
-benefits this provides. As the ES Module loader in Node.js enforces the use of
-[the full specifier path][], exporting the root rather than being explicit
-about entry is less expressive than either of the prior examples. Not only
-is encapsulation lost but module consumers are unable to
-`import feature from 'my-mod/feature'` as they need to provide the full
-path `import feature from 'my-mod/feature/index.js`.
+While the above supports good backwards-compatibility for package upgrades,
+a future major change for the package can then properly restrict the exports
+to only the specific feature exports exposed:
+
+```json
+{
+  "name": "my-package",
+  "exports": {
+    ".": "./lib/index.js",
+    "./feature/*.js": "./feature/*.js",
+    "./feature/internal/*": null
+  }
+}
+```
```

Comment:
I added this new section to note how the package can eventually be upgraded on a major. PTAL!

Reply:
Looks good to me. Do we want to include that we’re recommending `*.js` in order to produce non-bloated import maps, or is that better mentioned somewhere else?

Along with the first mention of import maps we could maybe also say that Node intends to support import maps in the future. It’s on the [loaders roadmap](https://github.com/nodejs/loaders), and putting it here in the docs would make it yet more official.

Reply:
I've added a note below and also included a dedicated section on import maps compatibility, linking to that.